addons/account/models/account_bank_statement.py

```diff
@@ -920,8 +920,8 @@ def process_reconciliation(self, counterpart_aml_dicts=None, payment_aml_rec=Non
 
             # Create The payment
             payment = self.env['account.payment']
+            partner_id = self.partner_id if self.partner_id else aml_dict['move_line'].partner_id if aml_dict.get('move_line') else False
```

Comment:
Seems simpler to do: `self.partner_id or (aml_dict.get('move_line') and aml_dict['move_line'].partner_id) or self.env['res.partner']`